File: CK3toEU4/Source/Mappers/TraitScraper/TraitScraping.cpp
#include "TraitScraping.h"
#include "ParserHelpers.h"

mappers::TraitScraping::TraitScraping(std::istream& theStream)
{
	registerKeys();
	parseStream(theStream);
	clearRegisteredKeywords();
}

void mappers::TraitScraping::registerKeys()
{
	registerKeyword("index", [this](const std::string& personalityName, std::istream& theStream) {
		index = commonItems::singleInt(theStream).getInt();
	});
	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
}

File: CK3ToEU4/Source/Mappers/TraitScraper/TraitScraping.cs
using commonItems;

namespace CK3ToEU4.Mappers.TraitScraper;

class TraitScraping
{
	public TraitScraping() {}
	public TraitScraping(std::istream& theStream)
	{
		var parser = new Parser();
		registerKeys(parser);
		parser.ParseStream(reader);
	}

	public int getIndex() { return index; }

	private void registerKeys(Parser parser)
	{
		parser.RegisterKeyword("index", [this](const std::string& personalityName, std::istream& theStream) {
			index = commonItems::singleInt(theStream).getInt();
		});
		parser.RegisterRegex(commonItems::catchallRegex, commonItems::ignoreItem);
	}

	private int index = 0;
};